### PR TITLE
CANS-857 Rework AuthBoundary front end component to use metadata permissions instead of Security API

### PR DESCRIPTION
**Pull request by vladimirPe**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-857

## Description
AuthBoundary was deleted as a redundant component, now all authorization information is a part of an entity (client or assessment)
All related code was reworked.

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by vladimirPe** [comment]
jenkins retest this please

**Comment by LeonidMarushevskyi** [comment]
jenkins retest this please

**Comment by LeonidMarushevskyi** [comment]
Jenkins retest this please

**Comment by nasar-cambria** [comment]
jenkins re-test this please

**Comment by zliuwd** [comment]
retest this please

**Comment by zliuwd** [comment]
retest this please

**Comment by vladimirPe** [comment]
jenkins retest this please

File: app/javascript/Application/components/Client/AssessmentHistory/ClientAssessmentHistory.js [hunk 1]
```diff
@@ -84,15 +74,14 @@ class ClientAssessmentHistory extends PureComponent {
 
 ClientAssessmentHistory.propTypes = {
   assessments: PropTypes.arrayOf(PropTypes.object),
-  clientIdentifier: PropTypes.string,
+  client: PropTypes.object.isRequired,
```
Comment (wmitchell-cambria) [inline app/javascript/Application/components/Client/AssessmentHistory/ClientAssessmentHistory.js]: This only validates that the client is an `object`, not that it has any required shape for the `isAuthorized` check. If there are specific fields needed here, then we can get early and clear error messages by specifying more shape.

**Pull request merged**